### PR TITLE
Adds @RequiresCRUD security annotation && Interceptors

### PR DESCRIPTION
**Pull request by Sherpard**
Smart Annotation that takes care of adding a permission check, 
User is able to configure how to resolve the CRUD verb implementing `CRUDActionResolver` on their code

### CODE REVIEW

**Comment by adrienlauer** [comment]
Can you show a simple example of how to use this annotation ? I'm somewhat understanding its purpose from the test, but an explanation would help :)

File: security/core/src/it/java/org/seedstack/seed/security/internal/fixtures/TestActionResolver.java [hunk 1]
```diff
@@ -22,34 +19,29 @@
 
 public class TestActionResolver implements CRUDActionResolver {
```
Comment (ejemba) [inline security/core/src/it/java/org/seedstack/seed/security/internal/fixtures/TestActionResolver.java]: I'd name the Class CrudActionResolver in order to be compliant to java standard name with acronyms and also to be flat with CrudSecurityIT. 
I also would rename CRUDAction to CrudAction for the same reasons.

Comment (adrienlauer) [reply security/core/src/it/java/org/seedstack/seed/security/internal/fixtures/TestActionResolver.java]: Agree. I suggested those names but it's better to keep CamelCase even for acryonyms.

Comment (Sherpard) [reply security/core/src/it/java/org/seedstack/seed/security/internal/fixtures/TestActionResolver.java]: Agree, already commited the cangue :)

**Comment by adrienlauer** [comment]
The canResolve method is not useful because resolve already returns an Optional and can return empty if nothing is detected. I removed it and refactored the implementation of the interceptors a bit in my fork at https://github.com/adrienlauer/seed/tree/feature/RequiresRestAnnotation.

**Pull request merged**